engine/src/main/java/org/pentaho/di/trans/steps/pgbulkloader/PGBulkLoader.java

```diff
@@ -133,12 +136,40 @@ public String getCopyCommand( ) throws KettleException {
     return contents.toString();
   }
 
+  void checkClientEncoding() throws Exception {
+    Connection connection = data.db.getConnection();
+
+    Statement statement = connection.createStatement();
+
+    try {
+      try (ResultSet rs = statement.executeQuery("show client_encoding")) {
+        if (!rs.next()) {
+          throw new KettleException( "Cannot detect client encoding" );
+        }
+
+        if (rs.getMetaData().getColumnCount() != 1) {
+          throw new KettleException( "Cannot detect client encoding" );
+        }
+
+        String clientEncodingStr = rs.getString(1);
+        logBasic( "Detect client encoding: " + clientEncodingStr );
+        clientEncoding = Charset.forName(clientEncodingStr);
+      }
+    } catch ( Exception ex ) {
+      throw new KettleException( "Error while checking client encoding", ex );
+    } finally {
+      statement.close();
+    }
+  }
+
   private void do_copy( PGBulkLoaderMeta meta, boolean wait ) throws KettleException {
     data.db = getDatabase( this, meta );
     String copyCmd = getCopyCommand();
     try {
       connect();
 
+      checkClientEncoding();
```

Comment:
After this line, you need to be sure that clientEncoding is non-null - if it is, assign to the system default encoding so that the system behaves as it always has in the event of a failure to detect the encoding of the client connection.